src/ring_theory/ideal/nc_jacobson_radical.lean

```diff
@@ -0,0 +1,143 @@
+-/
+def nc_jacobson (R : Type u) [ring R] : ideal R := jacobson ⊥
+
+lemma nc_jacobson_def : nc_jacobson R = Inf {I : ideal R | I.is_maximal } :=
+congr_arg Inf $ by simp
+
+lemma has_left_inv_one_add_of_mem_jacobson {x : R} :
+  x ∈ nc_jacobson R → has_left_inv (1 + x):=
+begin
+  contrapose,
+  rw not_has_left_inv_iff_mem_maximal,
+  rintro ⟨I, hImax, hxxI⟩ hx,
+  refine hImax.ne_top _,
+  rw eq_top_iff_one,
+  have hxI : x ∈ I := by { rw [nc_jacobson_def, mem_Inf] at hx, apply hx hImax },
+  exact (add_mem_cancel_right hxI).mp hxxI,
+end
+
+lemma one_add_mul_self_mem_maximal_of_not_mem_maximal {x : R} {I : ideal R}
+  (h₁ : I.is_maximal) (h₂ : x ∉ I) : ∃ a : R, 1 + a * x ∈ I :=
+begin
+  have : (1 : R) ∈ I ⊔ span {x},
+  { rw is_maximal_iff at h₁,
+    apply h₁.2 _ _ le_sup_left h₂,
+    apply mem_sup_right,
+    apply submodule.mem_span_singleton_self },
+  rw submodule.mem_sup at this,
+  obtain ⟨m, hmI, y, hy, hmy⟩ := this,
+  rw mem_span_singleton' at hy,
+  obtain ⟨a, rfl⟩ := hy,
+  existsi -a,
+  rwa [←hmy, neg_mul, add_neg_cancel_right],
+end
+
+/-- Characterizations of the Jacobson radical of a ring.
+
+The following are equivalent for an element `x` in a ring `R`.
+* 0: `x` is in the Jacobson radical of `R`, that is, contained in every maximal left ideal.
+* 1: `1 + a * x` has a left inverse for any `a : R`.
+* 2: `1 + a * x` is a unit for any `a : R`.
+* 3: `1 + x * b` is a unit for any `b : R`.
+* 4: `1 + a * x * b` is a unit for any `a b : R`.
+-/
+theorem mem_nc_jacobson_tfae {R : Type u} [ring R] (x : R) : tfae [
+  x ∈ nc_jacobson R,
+  ∀ a : R, has_left_inv (1 + a * x),
+  ∀ a : R, is_unit (1 + a * x),
+  ∀ b : R, is_unit (1 + x * b),
+  ∀ a b : R, is_unit (1 + a * x * b)] :=
+begin
+  tfae_have : 1 → 2,
+  { exact λ hx a, has_left_inv_one_add_of_mem_jacobson $
+    (nc_jacobson R).smul_mem' a hx },
+  tfae_have : 2 → 3,
+  { intros hx a,
+    obtain ⟨c, hc⟩ := hx a,
+    apply is_unit_of_left_inv_of_has_left_inv hc,
+    suffices : c = 1 + ( -c * a * x),
+    { rw this, apply hx },
+    { calc c = c * (1 + a * x) + ( -c * a * x) : by noncomm_ring
+        ...  = 1 + ( -c * a * x) : by rw hc } },
+  tfae_have : 3 → 5,
+  { intros hx _ _,
+    apply is_unit.one_add_mul_swap,
+    rw ←mul_assoc,
+    apply hx },
+  tfae_have : 5 → 1,
+  { intro h,
+    by_contra hx,
+    rw [nc_jacobson_def, submodule.mem_Inf] at hx,
+    simp only [not_forall] at hx,
+    rcases hx with ⟨I, hImax, hxI⟩,
+    refine hImax.ne_top _,
+    obtain ⟨a, ha⟩ := one_add_mul_self_mem_maximal_of_not_mem_maximal hImax hxI,
+    apply eq_top_of_is_unit_mem _ ha,
+    specialize h a 1,
+    rwa [mul_assoc, mul_one] at h },
+  tfae_have : 3 ↔ 4,
+  { split; exact λ h b, (h b).one_add_mul_swap },
+  tfae_finish,
+end
+
+open mul_opposite
+/-- The Jacobson radical of `R` coincides with that of its opposite ring `Rᵐᵒᵖ`. -/
+theorem nc_jacobson_op {x : R} :
+  x ∈ nc_jacobson R ↔ op x ∈ nc_jacobson Rᵐᵒᵖ :=
```

Comment:
Can you you turn this into a statement about `jacobson`?

Reply:
Sorry I don't understand what you say. Is `I` a two-sided ideal, and do you mean that intersection of left ideals containing `I` coincides with that of right ideals?
If so, it's correct, but in this case I have to use quotient ring `R/I` (perhaps this could be avoided but lots of extra work needed), and in any case it will be better to do that in another PR.

Reply:
I think Eric is suggesting making statement about ideals rather than about the elements they contain. However I don't think this is possible for us until we introduce right and two-sided ideals.

[I do have a plan to make this happen though, e.g., #14399 and #14465]